Truncate Lengthy Tags on Root/Details Pages

zackcl opened the pull request:
Resolves #2420

This PR truncates tags longer than 16 characters and displays the full tag string in a tooltip when truncated. This prevents long tags from affecting the column width of the root table and keeps the layout clean.

Screenshots after change:

<img width="1000" alt="Screenshot 2025-04-22 at 1 58 09 AM" src="https://github.com/user-attachments/assets/82f2a80a-a317-436c-a8f8-18e9428e8964" />
<img width="1000" alt="Screenshot 2025-04-22 at 1 58 22 AM" src="https://github.com/user-attachments/assets/3e0ba17a-f6a2-4db0-aefd-44c0f77cc7ef" />

danoswaltCL reviewed (changes requested):
works of course but we'll have to remember to add each time in future when using mat-chip... this is prob the official way we want to handle long tag text throughout so we can add a common-tag component to wrap this logic really simply, i tried this out real quick to convince myself it works

```
@Component({
  selector: 'app-common-tag',
  standalone: true,
  imports: [CommonModule, MatTooltipModule, SharedModule],
  template: `
    <span
      class="chip-label"
      (click)="onTagClick()"
      [matTooltip]="tag.length > 16 ? tag : null"
      matTooltipPosition="above"
    >
      {{ tag | truncate : 16 }}
    </span>
  `,
})
export class CommonTagComponent {
  @Input() tag = '';
  @Output() tagClick = new EventEmitter<string>();

  onTagClick(): void {
    this.tagClick.emit(this.tag);
  }
}
```

```
// tagClick is an optional callback since we have that use-case
<app-common-tag 
            [tag]="tag" 
            (tagClick)="searchByTag($event)">
          </app-common-tag>
```

zackcl commented:
@danoswaltCL I added `CommonTagComponent` and `CommonTagListComponent` similar to what you suggested, but included the `mat-chip` element in the component, not just the `span`.

I also updated `CommonTagsInputComponent` to use truncated tags. I personally think it's much cleaner to enter long UUIDs into the Values field when the tags truncate to 16 characters (we can modify the character limit if needed).

danoswaltCL reviewed (approved):
cool makes sense to me. if no styles are attached to the component it can just be omitted and scss file deleted, no use in an empty file.

zackcl commented:
>  if no styles are attached to the component it can just be omitted and scss file deleted, no use in an empty file.

@danoswaltCL We currently have empty `scss` files everywhere in our codebase. I personally think it's OK to keep them and sometimes it's easier for me to quickly apply styles to the existing files. Maybe we could consider cleaning these up later once the component designs are more finalized.
